### PR TITLE
[FX-977] Add unit tests for refund flow

### PR DESCRIPTION
**Pull request by djoksimo**
## What

- [x] Add tests for entire refund codepath
- [x] Improve existing tests, reduce duplication and facilitate migration to vault submitter (instead of pin collectors)

## How

Can be released as-is

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt [hunk 1]
```diff
@@ -146,7 +146,14 @@ class ForageTerminalSDK(
         logger
             .addAttribute("payment_ref", paymentRef)
             .addAttribute("merchant_ref", merchantId)
-        logger.i("[POS] Call refundPayment for Payment $paymentRef with amount $amount for reason $reason on Terminal $posTerminalId")
+        logger.i(
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt:149]: The abbreviated version still shows the first part "[POS] Called refundPayment for Payment $paymentRef"

and the the rest is visible when you click on the log message in DD

image

File: forage-android/src/main/java/com/joinforage/forage/android/vault/VgsPinSubmitter.kt [hunk 2]
```diff
@@ -88,7 +88,7 @@ internal class VgsPinSubmitter(
         return try {
             // converting the response to a ForageApiError should throw
             // in the case of a VGS error
-            ForageApiError.ForageApiErrorMapper.from(errorResponse.body ?: "")
+            ForageApiError.ForageApiErrorMapper.from(errorResponse.body!!)
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/vault/VgsPinSubmitter.kt:91]: Nice catch! I'll revert this change

[the remark this answers is not in the document]
Comment (djoksimo) [reply forage-android/src/main/java/com/joinforage/forage/android/vault/VgsPinSubmitter.kt]: Done: 3379e9eb53113f0046b05f077b134eb3f31d6c94

File: forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dleis612) [reply forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt:149]: This is definitely an improvement well done.

**Review by dleis612 — APPROVED** [review]

**Pull request merged**